internal/expr/scan.go

```diff
@@ -117,29 +117,41 @@ func (re *ResultExpr) Decode(args ...any) (err error) {
 
 		v := reflect.ValueOf(arg)
 		if v.Kind() != reflect.Pointer {
-			return fmt.Errorf("none pointer paramter")
+			return fmt.Errorf("none pointer parameter")
 		}
 		v = reflect.Indirect(v)
 
 		re.decodeValue(v)
-
 	}
 
 	return nil
 }
 
 // decodeValue sets the fields in the reflected struct "v" which have tags
-// corrosponding to columns in current row of the query results.
+// corresponding to columns in current row of the query results.
 func (re *ResultExpr) decodeValue(v reflect.Value) error {
 	typeFound := false
 	for i, outDest := range re.outputs {
+		if outDest.typ.Kind() == reflect.Map {
```

Comment:
We could do a check on the type of the field instead of this, the we woulnd't need the `CheckValidMapType(outDest.typ)`. This should be safe since we will only have put a `mapKey` field if it really was a map, and it can't have changed.